kafka-connect-s3/src/main/java/io/confluent/connect/s3/TopicPartitionWriter.java

```diff
@@ -498,8 +517,19 @@ private void writeRecord(SinkRecord record) {
       endOffsets.put(currentEncodedPartition, 0L);
     }
 
-    RecordWriter writer = getWriter(record, currentEncodedPartition);
-    writer.write(record);
+    RecordWriter writer = getWriter(currentEncodedPartition);
+    try {
+      writer.write(record);
+    } catch (DataException e) {
+      if (reporter != null) {
+        reporter.report(record, e);
+        log.warn("Errant record written to DLQ due to: {}", e.getMessage());
+        writer.close();
```

Comment:
I know see we can't close here and advance the offsets. We'll have to keep the writer

Reply:
Updated, the actual writer won't be closed.